s3torchconnector/src/s3torchconnector/_s3_bucket_iterable.py

```diff
@@ -32,10 +32,9 @@ def __init__(self, client: S3Client, bucket: str, prefix: str):
         self._bucket = bucket
         self._list_stream = _PickleableListObjectStream(client, bucket, prefix)
 
-    def __iter__(self) -> Iterator[S3Reader]:
-        return map(
-            partial(self._client.from_bucket_and_object_info, self._bucket),
-            chain.from_iterable(map(_extract_object_info, self._list_stream)),
+    def __iter__(self) -> Iterator[S3BucketKeyData]:
```

Comment:
I understand why we need this change, but at the first glance it looks like we're creating a stronger coupling with S3BucketIterable, which we plan to simplify at some point. Is it worth to invest in that simplification now, rather than creating more dependencies on it?

Reply:
I think changing the signature here is fine, given that it's only used in one place and is a private class. Also, the previous usages were only to unpack the key/bucket out of the `S3Reader` so this requires less packing/unpacking. 

I do agree that we should consider simplifying this code at some point, though I'm not sure if this PR is the best time for it